src/device.rs

```diff
@@ -1965,6 +1965,18 @@ impl DeviceRef {
                                        options:options]
         }
     }
+    
+    pub fn new_buffer_from_slice<T:Copy>(
+        &self,
+        data:&[T],
+        options:MTLResourceOptions 
+    )-> Buffer {
+        self.new_buffer_with_data(
+            &data[0] as *const _ as *const std::ffi::c_void,
```

Comment:
Use `slice::as_ptr`, otherwise you won't get the correct provenance!

```suggestion
            data.as_ptr().cast::<std::ffi::c_void>(),
```